upstairs/src/lib.rs

```diff
@@ -2398,14 +2412,34 @@ impl Downstairs {
              * more to do here.  If it's a flush, then we want to be
              * sure to update the last flush for this client.
              */
-            if let IOop::Flush {
-                dependencies: _dependencies,
-                flush_number: _flush_number,
-                gen_number: _gen_number,
-                snapshot_details: _,
-            } = &job.work
-            {
-                self.ds_last_flush[client_id as usize] = ds_id;
+            match &job.work {
+                IOop::Flush {
+                    dependencies: _dependencies,
+                    flush_number: _flush_number,
+                    gen_number: _gen_number,
+                    snapshot_details: _,
+                } => {
+                    self.ds_last_flush[client_id as usize] = ds_id;
+                }
+                IOop::Read {
+                    dependencies: _dependencies,
+                    requests: _,
+                } => {
+                    /*
+                     * For a read, make sure the data from a previous read
+                     * has the same hash
+                     */
+                    let read_data: Vec<ReadResponse> = read_data.unwrap();
+                    assert!(!read_data.is_empty());
+                    if job.hashes != read_hashes {
```

Comment:
I think the variable name should change to `read_response_hashes`, it took me a sec to realize that this was a vec of hashes for each read response, not for each read